packages/model-viewer/src/features/scene-graph/api.ts

```diff
@@ -49,6 +49,32 @@ export declare interface Model {
    * @returns the first material to whose name matches `name`
    */
   getMaterialByName(name: string): Material|null;
+
+  /**
+   * Creates a new material variant from an existing material.
+   * @param originalMaterialIndex index of the material to clone the variant
+   *     from.
+   * @param materialName the name of the new material
+   * @param variantName the name of the variant
+   * @param activateVariant activates this material variant, i.e. the variant
+   *     material is rendered, not the existing material.
+   */
+  createVariantFromMaterial(
+      originalMaterialIndex: number, newMaterialName: string,
+      variantName: string, activateVariant: boolean): Material|null;
```

Comment:
this will actually ensure the variant is created and clone the material and assign the clone to the variant.
Using `createVariant` followed `setMaterialToVariant` is how to assign an existing material to a variant.

Reply:
ah, so this is more about cloning the material than creating the variant? Like it'll just create the variant if it doesn't happen to exist yet? Maybe a  name like `cloneMaterialIntoVariant`?

Reply:
That's right, done.

Reply:
changed to `createMaterialInstanceForVariant` and removed fail case.